controllers/generic.go

```diff
@@ -28,35 +26,39 @@ func LoadPreRenderData() {
 	defer close(stop)
 	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(cm interface{}) {
-			addToPrerenderData(cm.(*v1.ConfigMap))
+			AddToClusterData(cm.(*v1.ConfigMap))
 		},
 		UpdateFunc: func(oldCM interface{}, cm interface{}) {
-			addToPrerenderData(cm.(*v1.ConfigMap))
+			AddToClusterData(cm.(*v1.ConfigMap))
 		},
 		DeleteFunc: func(cm interface{}) {
-			delete(preRenderData, cm.(*v1.ConfigMap).Labels[ReferenceLabel])
+			delete(clusterData, cm.(*v1.ConfigMap).Labels[ReferenceLabel])
 		},
 	})
 	informer.Run(stop)
 }
 
-func addToPrerenderData(cm *v1.ConfigMap) {
+func AddToClusterData(cm *v1.ConfigMap) {
 	for k, v := range cm.Data {
-		if preRenderData[cm.Labels[ReferenceLabel]] == nil {
-			preRenderData[cm.Labels[ReferenceLabel]] = make(map[string]string)
+		if clusterData[cm.Labels[ReferenceLabel]] == nil {
+			clusterData[cm.Labels[ReferenceLabel]] = make(map[string]string)
 		}
-		preRenderData[cm.Labels[ReferenceLabel]][k] = v
+		clusterData[cm.Labels[ReferenceLabel]][k] = v
 	}
 }
 
-func GetPreRenderData(appLabels map[string]string) map[string]map[string]string {
-	for label, labelValue := range appLabels {
-		if strings.HasPrefix(label, expediaType) {
+func GetPreRenderData(labels map[string]string) map[string]map[string]string {
+	preRenderData := make(map[string]map[string]string)
+	for key, value := range clusterData {
+		preRenderData[key] = value
```

Comment:
So now clusterData is available to render the values ? Is there any specific 
field needed in that ? 
I think the field in question was `{{application.instance}}` which still should come from application labels.

Reply:
yea so I renamed prerenderdata to clusterdata as I can see it probably caused confusion. With this change the preRenderData will now have all the cluster level data and the application level data. The change was to copy the original prerenderdata(now named as clusterdata)  into another variable since the original clusterdata map was an instance variable and was supposed to hold only cluster level data and its updates from the informer. It shouldnt be updated with application details.

Reply:
Hope this made sense?

Reply:
I therefore did two things:
1) renamed prerenderdata to cluster data - to make it clear that it is cluster data
2) Copied the instance variable cluster data into a local variable and to this local variable, add the application data as well. This local variable will now serve the template rendering for all cluster and application values
So, when a new application is updated, then since its loading the local variable at runtime it will always have the release specific details for that application

Reply:
previously it was adding application details directly to the instance variable which lives permanently as long overwhelm is running. So if new applications are released that have missing labels (like release instance for e.g.,) will not update the instance variable and old data will be rendered into the new release which was one of the issues they were facing

Reply:
also theres no need to store all the application details in memory which was happening before